pkg/ingester/series.go

```diff
@@ -211,19 +211,21 @@ func (s *memorySeries) setChunks(descs []*desc) error {
 }
 
 type desc struct {
-	C          encoding.Chunk // nil if chunk is evicted.
-	FirstTime  model.Time     // Timestamp of first sample. Populated at creation. Immutable.
-	LastTime   model.Time     // Timestamp of last sample. Populated at creation & on append.
-	LastUpdate model.Time     // This server's local time on last change
-	flushed    bool           // set to true when flush succeeds
+	C          encoding.Chunk    // nil if chunk is evicted.
+	FirstTime  model.Time        // Timestamp of first sample. Populated at creation. Immutable.
+	LastTime   model.Time        // Timestamp of last sample. Populated at creation & on append.
+	LastUpdate model.Time        // This server's local time on last change
+	LastValue  model.SampleValue // The value of the last sample. Used for stale chunk flushing.
+	flushed    bool              // set to true when flush succeeds
 }
 
-func newDesc(c encoding.Chunk, firstTime model.Time, lastTime model.Time) *desc {
+func newDesc(c encoding.Chunk, firstTime model.Time, lastTime model.Time, lastValue model.SampleValue) *desc {
 	return &desc{
 		C:          c,
 		FirstTime:  firstTime,
 		LastTime:   lastTime,
 		LastUpdate: model.Now(),
+		LastValue:  lastValue,
```

Comment:
can you try to use `memorySeries.lastSampleValue` instead of adding a similar member here?